SCClassLibrary/Common/Streams/EventStreamCleanup.sc

```diff
@@ -1,47 +1,70 @@
+// See CallOnce help for details on argument handling
+
+CallOnce : Thunk {
+
+	value { |... args| ^this.prEvaluate(\valueArray, args) }
+	valueArray { |... args| ^this.prEvaluate(\valueArray, *args) }
```

Comment:
`*args` should be `args`, both for `valueArray` and `valueArrayEnvir`

Reply:
> *args should be args, both for valueArray and valueArrayEnvir

I don't think so.

- `value` gets arguments like `(2, 3)` and packs them into args = `[2, 3]`. Then valueArray spreads them out again.
- `valueArray` gets arguments like `[2, 3]` and packs them into args = `[ [2, 3] ]`. Then valueArray spreads out *one* layer and we need to unpack the other layer with `*`.

```
(
f = { |a, b| a * b };

f.value(2, 3).debug("f.value(2, 3)");
f.valueArray([2, 3]).debug("f.valueArray([2, 3])");

CallOnce(f).value(2, 3).debug("CallOnce value(2, 3)");
CallOnce(f).valueArray([2, 3]).debug("CallOnce valueArray([2, 3])");
)

-->

f.value(2, 3): 6
f.valueArray([2, 3]): 6
CallOnce value(2, 3): 6
CallOnce valueArray([2, 3]): 6
```

^^ This is correct behavior.

Reply:
ok, good. It would be a little more efficient to pass an array to `prEvaluate`. Then there is no need for the repeated expansion. But this is also fine.